internal/it/test_skip.go

```diff
@@ -0,0 +1,221 @@
+package it
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hazelcast/hazelcast-go-client/internal"
+)
+
+const (
+	skipHzVersion     = "hz"
+	skipClientVersion = "ver"
+	skipOS            = "os"
+	skipEnterprise    = "enterprise"
+	skipOSS           = "oss"
+	enterpriseKey     = "HAZELCAST_ENTERPRISE_KEY"
+)
+
+// SkipIf can be used to skip a test case based on comma-separated conditions.
+// The following conditions may be set:
+// "hz": Hazelcast version
+// "ver": go client version
+// "os": value of runtime.GOOS
+// "enterprise"/"oss": presence of enterprise key environment variable
+// Example: SkipIf(t, "ver > 1.1, hz = 5")
+func SkipIf(t *testing.T, conditionString string) {
+	skip, err := skipIf(conditionString)
+	if err != nil {
+		t.Fatalf(err.Error())
```

Comment:
If you change `skipIf`/`canSkip` to panic instead of returning an error, we can simplify the code here to:
```
if skip := canSkip(condition); skip {
    t.Skip("Skipping test: SkipIf conditions met")
}
```

Reply:
Addressed in 5071a2fabacf61a0c6750af893641b4dfc45f051